PowerEditor/src/NppIO.cpp

```diff
@@ -1741,23 +1741,40 @@ bool Notepad_plus::fileRename(BufferID id)
 		TCHAR *tabNewName = reinterpret_cast<TCHAR *>(strDlg.doDialog());
 		if (tabNewName)
 		{
-			success = true;
-			buf->setFileName(tabNewName);
-			bool isSnapshotMode = NppParameters::getInstance().getNppGUI().isSnapshotMode();
-			if (isSnapshotMode)
+			BufferID sameNamedBufferId = _pDocTab->findBufferByName(tabNewName);
+			if (sameNamedBufferId == BUFFER_INVALID)
+			{
+				sameNamedBufferId = _pNonDocTab->findBufferByName(tabNewName);
+			}
+			
+			if (sameNamedBufferId != BUFFER_INVALID)
 			{
-				generic_string oldBackUpFile = buf->getBackupFileName();
+				_nativeLangSpeaker.messageBox("RenameTabTemporaryNameAlreadyInUse",
+					_pPublicInterface->getHSelf(),
+					TEXT("The specified name is already in use."),

```

Comment:
Let's use explicit (dumb) one:
"The specified name is already in use on another tab."

Reply:
@donho 
What you want is how the FIRST commit is.
Can you just accept the first commit and ignore the second?

Reply:
@sasumner OK, cool.
I'll do a cherry pick.